src/uct/tcp/tcp_ep.c

```diff
@@ -257,37 +257,73 @@ unsigned uct_tcp_ep_progress_rx(uct_tcp_ep_t *ep)
     return recv_length > 0;
 }
 
+static inline ucs_status_t uct_tcp_ep_am_send(uct_tcp_iface_t *iface, uct_tcp_ep_t *ep,
+                                              const uct_tcp_am_hdr_t *hdr,
+                                              size_t check_length, const char *name)
+{
+    UCT_CHECK_LENGTH(hdr->length, 0, check_length, name);
```

Comment:
better check it before all other checks, i.e in the very beginning of in _short, _bcopy funcs